### PR TITLE
Fix: Renamed CreditNft and accessControl and added checks to cspell

### LINKED ISSUE #736: Rename `CreditNFT` to `CreditNft`
There are some [places](https://github.com/ubiquity/ubiquity-dollar/blob/896b25113b36bc1e18e3dd5bf03be88c9c7bb7e3/packages/contracts/src/dollar/upgradeInitializers/DiamondInit.sol#L38) in the code where `CreditNFT` naming is used. We have a convention to use it as `CreditNft` instead of `CreditNFT`.

There are also some places in the [code](https://github.com/ubiquity/ubiquity-dollar/blob/2642a8c42041f9577a5041ed74d8d83c7097e852/packages/contracts/src/dollar/core/CreditClock.sol#L16) where `accessCtrl` contraction is used. This contraction is unnecessary.

What should be done:
1. Rename all `CreditNFT` instances in the code to `CreditNft`
2. Check that `cspell` command in the pre-commit hook catches all `CreditNFT` namings and throws an error in this case
3. Rename all `accessCtrl` instance in the code to `accessControl`

### PR DESCRIPTION
**Pull request by baibhavKumar1**
Resolves #736

### CODE REVIEW

File: packages/dapp/components/utils/local-data.ts [hunk 1]
```diff
@@ -67,7 +67,7 @@ export async function fetchData() {
 
         console.log((await CreditNFTManager.connect(signer).getCreditNFTReturnedForDollars(signer.getAddress())).toBigInt(), "Credit NFT Returned Dollars");
```
Comment (molecula451) [inline packages/dapp/components/utils/local-data.ts:68]: CreditNft??

**Comment by rndquu** [comment]
@Wholesomebruh cspell workflow is [failing](https://github.com/ubiquity/ubiquity-dollar/actions/runs/5707906390/job/15465046760?pr=742), could you fix it?

**Comment by baibhavKumar1** [comment]
@rndquu , renamed creditNFT to creditNft. All checks are passing.

**Review by rndquu — CHANGES_REQUESTED** [review]
There still instances of `creditNFT*` in the code

- https://github.com/Wholesomebruh/ubiquity-dollar/blob/development/packages/contracts/src/dollar/libraries/LibCreditNftManager.sol#L54 (`creditNFTLengthBlocks => creditNftLengthBlocks`)
- https://github.com/Wholesomebruh/ubiquity-dollar/blob/development/packages/contracts/src/dollar/facets/CreditNftManagerFacet.sol#L24 (`setExpiredCreditNFTConversionRate => setExpiredCreditNftConversionRate`)
- etc...

To sum up all solidity files should be searched for `*CreditNFT*` or `*creditNFT*` and refactored to `*CreditNft*` or `*creditNft*`

Pls refactor

**Comment by baibhavKumar1** [comment]
@rndquu , is there anything more to change other than all the creditNFT and CreditNFT to creditNft and CreditNft respectively in the "solidity" files or is it required in "ts, js, jsx, tsx" files too? I will be searching for any word that has "credit" or "Credit" with "NFT", and change the NFT to Nft. Please state if there is anything else to add. Thank You.

**Comment by rndquu** [comment]
> @rndquu , is there anything more to change other than all the creditNFT and CreditNFT to creditNft and CreditNft respectively in the "solidity" files or is it required in "ts, js, jsx, tsx" files too? I will be searching for any word that has "credit" or "Credit" with "NFT", and change the NFT to Nft. Please state if there is anything else to add. Thank You.

Pls refactor only `.sol` files

**Comment by baibhavKumar1** [comment]
Sorry, just making changes to .sol results in build error. Need to recheck every changes. Close this PR.
